pages/studentPortal/labs/multiplication/game.tsx

```diff
@@ -1,14 +1,114 @@
-import MultiplicationBlock from "../../../../components/math/MultiplicationBlock";
-//create big box
+import React, { useState } from "react";
+import MultiplicationBlock, {
+  BlockState,
+} from "../../../../components/math/MultiplicationBlock";
+import { Button } from "../../../../components/ui/Button";
+
 export default function BlockComponentGallery() {
+  const [player, setPlayer] = useState(false);
+
+  function handlePlayer() {
+    setPlayer(!player);
+  }
+
+  const initialGameState = [
+    {
+      text: "72",
+      state: BlockState.NOT_SELECTED,
+      winner: -1,
+    },
+    {
+      text: "8x9",
+      state: BlockState.NOT_SELECTED,
+      winner: -1,
+    },
+    {
+      text: "8",
+      state: BlockState.NOT_SELECTED,
+      winner: -1,
+    },
+    {
+      text: "8x1",
+      state: BlockState.NOT_SELECTED,
+      winner: -1,
+    },
+    {
+      text: "16",
+      state: BlockState.NOT_SELECTED,
+      winner: -1,
+    },
+    {
+      text: "2x8",
+      state: BlockState.NOT_SELECTED,
+      winner: -1,
+    },
+  ];
+
+  const [gameState, setGameState] = useState(initialGameState);
+
+  function handleSelect(index) {
+    console.log("BLOCK WAS CLICKED: index ", index);
+    console.log(gameState[index].text);
+
+    let gameState2 = [...gameState];
+    if (player === true) {
+      gameState2[index].state = BlockState.PLAYER_ONE_SELECTED;
+      gameState2[index].winner = 1;
+    } else if (player === false) {
+      gameState2[index].state = BlockState.PLAYER_TWO_SELECTED;
+      gameState2[index].winner = 2;
+    }
+
+    setGameState(gameState2);
+  }
+
+  function handleReset() {
+    setGameState(initialGameState);
+  }
+  function longestSubarray(array, x) {
+    let maxlength = 0;
+    let sum = 0;
+    for (let i = 0; i < array.length; i++) {
+      let sum = 0;
+      if (array[i].winner == x) {
+        sum++;
+      } else {
```

Comment:
Empty else case can be removed

Reply:
It actually needed to contain let sum = 0 to reset the counter. I had it but must not have saved it lol So I added that.